packages/next/src/server/app-render/action-handler.ts

```diff
@@ -253,28 +253,41 @@ export async function handleAction({
           const {
             decodeReply,
             decodeReplyFromBusboy,
+            decodeAction,
           } = require(`react-server-dom-webpack/server.node`)
 
           if (isMultipartAction) {
-            const busboy = require('busboy')
-            const bb = busboy({ headers: req.headers })
-            req.pipe(bb)
+            if (isFetchAction) {
+              const busboy = require('busboy')
+              const bb = busboy({ headers: req.headers })
+              req.pipe(bb)
 
-            bound = await decodeReplyFromBusboy(bb, serverModuleMap)
+              bound = await decodeReplyFromBusboy(bb, serverModuleMap)
+            } else {
+              // React doesn't yet publish a busboy version of decodeAction
+              // so we polyfill the parsing of FormData.
+              const { Readable } = require('stream')
+              const UndiciRequest = require('undici').Request
+              const fakeRequest = new UndiciRequest('http://localhost', {
+                method: 'POST',
+                headers: { 'Content-Type': req.headers['content-type'] },
+                body: Readable.toWeb(req),
+                duplex: 'half',
+              })
+              const formData = await fakeRequest.formData()
+              const action = await decodeAction(formData, serverModuleMap)
+              await action()
+              // Skip the fetch path
+              return
+            }
           } else {
             const { parseBody } =
               require('../api-utils/node') as typeof import('../api-utils/node')
             const actionData = (await parseBody(req, '1mb')) || ''
 
-            if (isFormAction) {
-              actionId = actionData.$$id as string
-              if (!actionId) {
-                // Return if no action ID is found, it could be a regular POST request
-                return
-              }
+            if (isURLEncodedAction) {
```

Comment:
We don't ever use URL encoded bodies atm but if we ever did, it'd be equivalent of passing formData.